jbpm-system-test/jbpm-clustered-ejb-timers-test/src/test/java/org/kie/samples/integration/ClusteredEJBTimerSystemTest.java

```diff
@@ -0,0 +1,333 @@
+        String node = noCluster? nodeName : "node1";
+        try {
+             String content = FileUtils.readFileToString(new File(PREFIX_CLI_PATH+"template.cli"), "UTF-8");
+             content = content.replaceAll("%partition_name%", "\\\"ejb_timer_"+node+"_part\\\"");
+             File cliFile = new File(PREFIX_CLI_PATH+nodeName+".cli");
+             FileUtils.writeStringToFile(cliFile, content, "UTF-8");
+             cliFile.deleteOnExit();
+          } catch (IOException e) {
+             throw new RuntimeException("Generating file failed", e);
+          }
+    }
+
+    private static KieServicesClient authenticate(int port, String user, String password) {
+        String serverUrl = "http://localhost:" + port + "/kie-server/services/rest/server";
+        KieServicesConfiguration configuration = KieServicesFactory.newRestConfiguration(serverUrl, user, password);
+        
+        configuration.setTimeout(60000);
+        configuration.setMarshallingFormat(MarshallingFormat.JSON);
+        return  KieServicesFactory.newKieServicesClient(configuration);
+    }
+
+    private Long startProcessInNode1(String processName) {
+        Long processInstanceId = processClient1.startProcess(containerId, processName, singletonMap("id", "id1"));
+        assertNotNull(processInstanceId);
+        return processInstanceId;
+    }
+    
+    private Long startTaskInNode2(Long processInstanceId) {
+        List<TaskSummary> taskList = findReadyTasks(processInstanceId);
+
+        Long taskId = taskList.get(0).getId();
+        logger.info("Starting task {} on kieserver2", taskId);
+        taskClient2.startTask(containerId, taskId, DEFAULT_USER);
+        return taskId;
+    }
+
+    private List<TaskSummary> findReadyTasks(Long processInstanceId) {
+        List<String> status = Arrays.asList(Status.Ready.toString());
+        List<TaskSummary> taskList = taskClient2.findTasksByStatusByProcessInstanceId(processInstanceId, status, 0, 10);
+        return taskList;
+    }
+
+    private void abortProcess(KieServicesClient kieServicesClient, Long processInstanceId) {
+        QueryServicesClient queryClient = kieServicesClient.getServicesClient(QueryServicesClient.class);
+        
+        ProcessInstance processInstance = queryClient.findProcessInstanceById(processInstanceId);
+        assertNotNull(processInstance);
+        assertEquals(1, processInstance.getState().intValue());
+        processClient1.abortProcessInstance(containerId, processInstanceId);
+    }
+    
+    private static HikariDataSource getDataSource() {
+        HikariConfig hikariConfig = new HikariConfig();
+        hikariConfig.setJdbcUrl(postgreSQLContainer.getJdbcUrl());
+        hikariConfig.setUsername(postgreSQLContainer.getUsername());
+        hikariConfig.setPassword(postgreSQLContainer.getPassword());
+        hikariConfig.setDriverClassName(postgreSQLContainer.getDriverClassName());
+        
+        return new HikariDataSource(hikariConfig);
+    }
```

Comment:
Is there any reason a [`PoolingDataSource`](https://github.com/kiegroup/jbpm/blob/4b7c47df6cee60cffa5fe4c3a41ad1a8c9504352/jbpm-test-util/src/main/java/org/jbpm/test/persistence/util/PersistenceUtil.java#L183-L186) wasn't used?

Reply:
No, just a example with another client, it could be changed in other iteration

Reply:
Sure, no problem.